python/composio/tools/base/abs.py

```diff
@@ -426,6 +426,18 @@ def setup_children(obj: t.Type["Tool"]) -> None:
             if hasattr(obj, "logo"):
                 setattr(trigger, "logo", getattr(obj, "logo"))
 
+    @staticmethod
+    def _get_description(obj) -> str:
+        return " ".join(
+            filter(
+                bool,
+                map(
+                    str.strip,
+                    (obj.__doc__ if obj.__doc__ else obj.display_name).splitlines(),
```

Comment:
Consider adding error handling for cases where neither `__doc__` nor `display_name` exists to prevent potential AttributeError:

```python
description_source = obj.__doc__ if hasattr(obj, "__doc__") and obj.__doc__ else (
    obj.display_name if hasattr(obj, "display_name") else ""
)
return " ".join(filter(bool, map(str.strip, description_source.splitlines())))
```